job-server/src/spark.jobserver/auth/SJSAuthenticator.scala

```diff
@@ -0,0 +1,120 @@
+package spark.jobserver.auth
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+import spray.routing.directives.AuthMagnet
+import spray.routing.authentication.UserPass
+import spray.routing.authentication.BasicAuth
+
+import spray.routing.authentication._
+import spray.routing.directives.AuthMagnet
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import org.slf4j.LoggerFactory
+import com.typesafe.config.Config
+import org.apache.shiro.SecurityUtils
+import org.apache.shiro.authc._
+import org.apache.shiro.config.IniSecurityManagerFactory
+import org.apache.shiro.mgt.SecurityManager
+import org.apache.shiro.util.Factory
+import org.apache.shiro.subject.Subject
+
+trait SJSAuthenticator {
+
+  def asShiroAuthenticator(config: Config)(implicit ec: ExecutionContext): AuthMagnet[AuthInfo] = {
+    val logger = LoggerFactory.getLogger(getClass)
+
+    //TODO - can we read this information from the config instead?
+    val ldapFactory = new IniSecurityManagerFactory(config.getString("shiro.path.ini"))
+    val sManager = ldapFactory.getInstance()
+    SecurityUtils.setSecurityManager(sManager)
+
+    val allowedGroups: Option[java.util.List[String]] = {
+      if (config.hasPath("shiro.ldap.allowedGroups")) {
+        val g = config.getStringList("shiro.ldap.allowedGroups")
+        if (g.size() > 0) { Some(g) } else { None }

```

Comment:
@velvia Well, I think yes, it needs to be wrapped in an Option as we need to be able distinguish between the case of an empty group list (meaning that no users are allowed) and the case when no groups are set at all (meaning that we don't want to check on groups). The alternative would be a separate boolean variable that indicates whether group authentication is requested at all.